api/app.py

```diff
@@ -0,0 +1,151 @@
+from flask import Flask, request, jsonify
+from flask_executor import Executor
+from git import Repo
+from wtforms import Form, StringField, IntegerField, validators
+import json
+import shutil
+import subprocess
+
+app = Flask(__name__)
+executor = Executor(app)
+
+
+class ResultsGetForm(Form):
+    job_hash = StringField('job_hash', [validators.InputRequired()])
+
+
+class DataForm(Form):
+    url = StringField('url', [validators.InputRequired()])
+    limit = IntegerField('limit', default=1)
+    step = IntegerField('step', default=1)
+    branch = StringField('branch')
+    commit = StringField('commit')
+
+
+"""
+    Method to clone the provided repo and start the executor to run the cloc job before returning the STARTED status and the hash of the current job 
+"""
+
+
+def start_runner(form, runner_type):
+    url, job_hash = form.url.data, hash(form)
+    clone(url, job_hash, runner_type=runner_type)
+
+    if runner_type == "single":
+        executor.submit_stored(
+            f'cloc_results_{job_hash}', execute_cloc, job_hash, runner_type
+        )
+    elif runner_type == "history":
+        executor.submit_stored(
+            f'cloc_results_{job_hash}', execute_cloc, job_hash, runner_type, limit=form.limit.data, step=form.step.data
+        )
+    return jsonify({'status': 'STARTED', 'hash': job_hash})
+
+
+"""
+    Clones the repo to a tmp folder based on the hash of the form input
+"""
+
+
+def clone(url, job_hash, **kwargs):
+    path = f"/tmp/cloc-api-{job_hash}"
```

Comment:
Is there a chance this could introduce a security vulerability? Does the user have any control over the `job_hash` string? This could potentially allow them to access another file than the one intended.